### PR TITLE
Review Activity Log

### PR DESCRIPTION
**Pull request by Rownaknira**

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -38,6 +38,7 @@ gem "rmagick"
 gem "faker", "~> 1.6", ">= 1.6.1"
 gem "cancancan", "~> 1.13", ">= 1.13.1"
 gem "file_validators", "~> 2.0", ">= 2.0.2"
+gem 'public_activity'
```
Comment (abdullahallmehedi) [inline Gemfile]: ""

File: app/decorators/review_decorator.rb [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+  # Define presentation-specific methods here. Helpers are accessed through
+  # `helpers` (aka `h`). You can override attributes, for example:
+  #
+  #   def created_at
+  #     helpers.content_tag :span, class: 'time' do
+  #       object.created_at.strftime("%a %m/%d/%y")
+  #     end
+  #   end
+
+end
```
Comment (abdullahallmehedi) [inline app/decorators/review_decorator.rb]: remove this file

File: app/models/review.rb [hunk 3]
```diff
@@ -1,5 +1,14 @@
...
+  include PublicActivity::Model
+  tracked owner: ->(controller, model) {controller && controller.current_user}
   belongs_to :user
   belongs_to :book
   has_many :comments, dependent: :destroy
+
+  class << self
+    def review_acitvities
+      PublicActivity::Activity.order("created_at desc").
+        where(trackable_type:"Review")
```
Comment (abdullahallmehedi) [inline app/models/review.rb]: spacing

File: app/views/reviews/_review_activity_log.html.erb [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+        <tr>
+          <td>
+            <%= t ".review_content" %>
+            <%= review_activity.trackable.content %>
+            <% key = review_activity.key %>
+            <%= key[7,key.length] << ".by" %>
+            <% first_name = review_activity.owner.first_name %>
+            <% last_name = review_activity.owner.last_name %>
+            <%= link_to "#{first_name+" "+last_name}", current_user %>
+            <%= time_ago_in_words(review_activity.created_at) %>
```
Comment (abdullahallmehedi) [inline app/views/reviews/_review_activity_log.html.erb]: ()

**Comment by abdullahallmehedi** [comment]
ok

**Comment by noor-framgia** [comment]
ok

**Comment by SSalekin** [comment]
ok

**Pull request merged**